### PR TITLE
[core] Add database name and table name in BucketsTable

### PR DESCRIPTION
**Pull request by LsomeYeah**
### Purpose

Linked issue: close #xxx

- To specify the database of the wrapped table of  `BucketsTable`;
- add two columns (databaseName and tableName) for `BucketsTable `;

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/table/system/BucketsTable.java [hunk 1]
```diff
@@ -67,9 +71,23 @@ public class BucketsTable implements DataTable, ReadonlyTable {
...
+        this.tableName = wrapped.name();
+    }
+
+    // if need to specify the database of a table, use this method
+    public BucketsTable(
+            FileStoreTable wrapped, boolean isContinuous, String databaseName, String tableName) {
+        this.wrapped = wrapped;
+        this.isContinuous = isContinuous;
+        this.databaseName = databaseName;
+        this.tableName = tableName;
```
Comment (yuzelin) [inline paimon-core/src/main/java/org/apache/paimon/table/system/BucketsTable.java:90]: this(...)
tableName is not necessary to be an argument.

File: paimon-core/src/main/java/org/apache/paimon/table/system/BucketsTable.java [hunk 2]
```diff
@@ -99,6 +117,9 @@ public RowType rowType() {
         fields.add(new DataField(1, "_PARTITION", newBytesType(false)));
         fields.add(new DataField(2, "_BUCKET", new IntType(false)));
         fields.add(new DataField(3, "_FILES", newBytesType(false)));
+        fields.add(
+                new DataField(4, "_DATABASE_NAME", new VarCharType(Integer.MAX_VALUE).nullable()));
+        fields.add(new DataField(5, "_TABLE_NAME", new VarCharType(Integer.MAX_VALUE)));
```
Comment (yuzelin) [inline paimon-core/src/main/java/org/apache/paimon/table/system/BucketsTable.java:122]: Use `VarCharType(boolean isNullable, int length)`

Comment (LsomeYeah) [reply paimon-core/src/main/java/org/apache/paimon/table/system/BucketsTable.java]: ok

**Review by yuzelin — APPROVED** [review]
LGTM, waiting for test result.

**Review by yuzelin — APPROVED** [review]
+1

**Pull request merged**